state/state.go

```diff
@@ -19,13 +19,9 @@ type State interface {
 }
 
 type Snapshot interface {
-	Get(k []byte) ([]byte, bool)
-	Commit(objs []*Object) (Snapshot, []byte)
-}
+	readSnapshot
```

Comment:
why do we need a private interface instead of raw type?

Reply:
The idea is to abstract the transition execution (i.e. `evm`) from the state model (`itrie`).

Reply:
why do we need to abstract, do we have a few implementations? or just for testing?

Reply:
So this `Snapshot` object as it right now does two things: `readSnapshot` + `Commit`. It is like that because the `executor` object does the `EVM` processing (with `readSnapshot`) and computes the trie (with `Commit`).

However, I would like to move towards an architecture where those two things are separated from each other (in different packages) in order to support things like the Erigon flat data model (where commit does not happen on each block).

The `readSnapshot` is an iteration towards that model. It is also easier to test. I can leave it as it is right now since there is no need for it and introduce the change later as the use of the interface becomes more clear.